sui/src/validator.rs

```diff
@@ -41,10 +39,11 @@ async fn main() -> Result<(), anyhow::Error> {
     let guard = telemetry_subscribers::init(config);
 
     let cfg = ValidatorOpt::parse();
-    let genesis_conf: GenesisConfig = PersistedConfig::read(&cfg.genesis_config_path)?;
-    let address = cfg.address;
 
-    let (network_config, _, _) = genesis(genesis_conf).await?;
+    let network_config_path = sui_config_dir()?.join(SUI_NETWORK_CONFIG);
```

Comment:
Its looks like you're expecting the genesis to be pre-configured and the store sent over the wire?
Can you explain this approach?

The plan we discussed was to specify the genesis config and each validator configures itself

Reply:
Yes, we're still doing that, just in a slightly different way. Please see https://github.com/MystenLabs/devnet/blob/mlogan/devnet/docker/validator/entry.sh#L6

Reply:
Okay my approach was built in such a way that you only needed validator binary
With your approach, you need sui binary also for genesis.
Why not combine them so the validator binary can either do genesis or load existing?
This also speeds up container build because we only need to ship/build validator binary